Source/Core/Core/AchievementManager.cpp

```diff
@@ -239,6 +239,37 @@ void AchievementManager::DoFrame()
   }
 }
 
+bool AchievementManager::CanPause()
+{
+  u32 frames_to_next_pause = 0;
+  bool can_pause = rc_client_can_pause(m_client, &frames_to_next_pause);
+  if (!can_pause)
+  {
+    OSD::AddMessage("Cannot spam pausing in hardcore mode.", OSD::Duration::VERY_LONG,
+                    OSD::Color::RED);
+    OSD::AddMessage(
+        fmt::format("Can pause in {} seconds.", static_cast<float>(frames_to_next_pause) / 60),
+        OSD::Duration::VERY_LONG, OSD::Color::RED);
+  }
+  return can_pause;
+}
+
+void AchievementManager::DoIdle()
+{
+  if (!IsGameLoaded())
+    return;
+
+  m_idle_thread = new std::thread([]() {
```

Comment:
This is nonsense. Where do I even start here.

- This thread is never joined, so it will just live forever.
- It is also never deleted, leaking memory.
- Every call to DoIdle() will create a new thread even if one already exists.
- Nothing in the thread synchronizes with the outside world. `m_client` may become invalid at any time and you'd have no idea.

Reply:
> This thread is never joined, so it will just live forever.

Doesn't it stop executing within a second or so of the state becoming Paused?

I very much agree regarding the memory leak and the validity of m_client, though. (The validity of m_system could possibly also become relevant in the future.)